src/core/mac/sub_mac.cpp

```diff
@@ -222,6 +231,44 @@ otError SubMac::Send(void)
     return error;
 }
 
+void SubMac::ProcessTransmitSecurity(void)
+{
+    const ExtAddress *extAddress            = NULL;
+    bool              processTransmitAesCcm = true;
+    uint8_t           keyIdMode;
+
+    VerifyOrExit(ShouldHandleTransmitSecurity(), OT_NOOP);
+    VerifyOrExit(mTransmitFrame.GetSecurityEnabled(), OT_NOOP);
+    VerifyOrExit(!mTransmitFrame.IsSecurityProcessed(), OT_NOOP);
+
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    if (mTransmitFrame.GetTimeIeOffset() != 0)
+    {
+        // Transmit security will be processed after time IE content is updated.
+        processTransmitAesCcm = false;
+    }
+#endif
```

Comment:
```suggestion
#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
VerifyOrExit(mTransmitFrame.GetTimeIeOffset() == 0);
#endif
```
And move to the line just before `mTransmitFrame.ProcessTransmitAesCcm(*extAddress);`